lib/logdna/client.rb

```diff
@@ -125,14 +120,20 @@ def flush
           send_request
         end
         @lock.unlock
+        Thread.current.exit
       else
         schedule_flush
       end
     end
 
     def exitout
-      flush
-      @internal_logger.debug("Exiting LogDNA logger: Logging remaining messages")
+      if @buffer.any? || @side_messages.any?
+        Thread.new {
+          flush
+          @internal_logger.debug("Exiting LogDNA logger: Logging remaining messages")
```

Comment:
Should the logging message come before the flush instead of after?

Reply:
Agree with @jakedipity - I think, specifically this message should come in the beginning of `exitout` or just before `flush`